### PR TITLE
Fix shape property for hand

### PR DESCRIPTION
**Pull request by sergiik**

### CODE REVIEW

File: poker/hand.py [hunk 1]
```diff
@@ -191,7 +191,12 @@ def is_pair(self):
 
     @property
     def shape(self):
-        return Shape(self._shape)
+        if self.is_pair:
+            return Shape.PAIR
+        elif self.is_suited:
+            return Shape.SUITED
+        else:
+            return Shape.OFFSUIT
```
Comment (kissgyorgy) [inline poker/hand.py]: What was wrong with the original implementation? This seems much more complicated to me.

File: poker/strategy.py [hunk 2]
```diff
@@ -44,6 +44,8 @@ def __init__(self, strategy, source='<string>'):
         for name in self._config.sections():
             # configparser set non-specified values to '', we want default to None
             values = dict.fromkeys(_Situation.__slots__, None)
+            if '__weakref__' in values:
+                del values['__weakref__']
             for key, val in self._config[name].items():
```
Comment (kissgyorgy) [inline poker/strategy.py]: Why is this needed?

Comment (sergiik) [reply poker/strategy.py]: Without this change py.test fails
===================================================== ERRORS ======================================================
_________________________________ ERROR collecting tests/strategy/test_strategy.py _________________________________
tests/strategy/test_strategy.py:12: in <module>
    strategy = Strategy.from_file(filedir / 'push.strategy')
poker/strategy.py:64: in from_file
    return cls(strategy, source=filename)
poker/strategy.py:56: in __init__
    self._situations[name] = _Situation(**values)
E   TypeError: __init__() got an unexpected keyword argument '__weakref__'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================= 1 error in 0.48 seconds =========================================

File: poker/hand.py (hunk 1 above)
Comment (sergiik) [reply poker/hand.py]: With original implementation the following simple code fails:

$ cat test.py 
from poker.hand import Range, Hand

range = Range("AK")
hand = Hand('AKs')
if hand in range:
    print "In range"
else:
    print "Out of range"


$ python test.py
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    if hand in range:
  File "/usr/local/lib/python2.7/dist-packages/poker/hand.py", line 627, in __contains__
    return item in self._all_hands
  File "/usr/local/lib/python2.7/dist-packages/poker/hand.py", line 101, in __hash__
    return hash(self.first) + hash(self.second) + hash(self.shape)
  File "/usr/local/lib/python2.7/dist-packages/poker/hand.py", line 194, in shape
    return Shape(self._shape)
  File "/usr/local/lib/python2.7/dist-packages/poker/_common.py", line 28, in __call__
    return super(_PokerEnumMeta, cls).__call__(value)
  File "/usr/lib/python2.7/dist-packages/enum/__init__.py", line 347, in __call__
    return cls.__new__(cls, value)
  File "/usr/lib/python2.7/dist-packages/enum/__init__.py", line 662, in __new__
    raise ValueError("%s is not a valid %s" % (value, cls.__name__))
ValueError: s is not a valid Shape

Comment (kissgyorgy) [reply poker/hand.py]: What version of Python are you using?

Comment (sergiik) [reply poker/hand.py]: $ python --version
Python 2.7.15+

The same error happened here in travis 
https://travis-ci.org/pokerregion/poker/builds/572083053

**Comment by kissgyorgy** [comment]
I'm converting the hole library to Python 3 instead of fixing this bug. Would that be ok for you?

**Comment by sergiik** [comment]
yes that should be fine for me

**Comment by kissgyorgy** [comment]
@sergiik Can you try `master` branch with Python 3.6+ ? If it works for you, I will publish a new release.

**Comment by sergiik** [comment]
I tested master branch with Python 3.6.8 and it looks fine for me. Thanks

**Comment by kissgyorgy** [comment]
You can install it: https://pypi.org/project/poker/0.30.0/

**Comment by sergiik** [comment]
Thanks
